jar-connector/src/main/java/com/sixsq/slipstream/connector/UsageRecorder.java

```diff
@@ -75,9 +75,8 @@ public static void insertEnd(String instanceId, String user, String cloud, List<
 			}
 
 			if(!hasRecorded(cloud, instanceId)){
-				logger.fine("Not recorded => avoiding inserting usage record END for "
+				logger.fine("Not recorded but still inserting usage record END for "
 						+ metrics + ", " + describe(instanceId, user, cloud));
-				return;
```

Comment:
What happens if we really hasn't opened a recorded for `cloud, instanceId` because we have never seen this before?  I.e. the provided pair `cloud, instanceId` for inserting the end is an artefact.  So, we have an `end` but no `start`.